### ISSUE TITLE
`update --all` channel priority bogus

### ISSUE THREAD

**Issue by pitrou** [conda/conda#3303]
```
(35) Z:\numba>conda config --get channels
--add channels 'defaults'   # lowest priority
--add channels 'numba'
--add channels 'antoine'   # highest priority

(35) Z:\numba>conda update --all
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment C:\Miniconda3\envs\35:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    llvmlite-0.12.1            |           py35_0         5.8 MB
    ipython-5.1.0              |           py35_0         1.0 MB
    ------------------------------------------------------------
                                           Total:         6.8 MB

The following packages will be UPDATED:

    ipython:  5.0.0-py35_0             --> 5.1.0-py35_0

The following packages will be SUPERCEDED by a higher-priority channel:

    llvmlite: 0.13.0.dev-py35_61 numba --> 0.12.1-py35_0

Proceed ([y]/n)? n

Exiting


(35) Z:\numba>conda update llvmlite
Fetching package metadata .........
# All requested packages already installed.
# packages in environment at C:\Miniconda3\envs\35:
#
llvmlite                  0.13.0.dev              py35_61    numba
```

**Comment by pitrou** [comment]
Most bemusing thing is that, if invert the channel order, the result is exactly the same:
```
(35) Z:\numba>conda config --get channels
--add channels 'antoine'   # lowest priority
--add channels 'numba'
--add channels 'defaults'   # highest priority

(35) Z:\numba>conda update --all
Fetching package metadata ...Uncaught backoff with errno 2
......
Solving package specifications: ..........

Package plan for installation in environment C:\Miniconda3\envs\35:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    llvmlite-0.12.1            |           py35_0         5.8 MB
    ipython-5.1.0              |           py35_0         1.0 MB
    ------------------------------------------------------------
                                           Total:         6.8 MB

The following packages will be UPDATED:

    ipython:  5.0.0-py35_0             --> 5.1.0-py35_0

The following packages will be SUPERCEDED by a higher-priority channel:

    llvmlite: 0.13.0.dev-py35_61 numba --> 0.12.1-py35_0

Proceed ([y]/n)? n

Exiting


(35) Z:\numba>conda update llvmlite
Fetching package metadata ...Uncaught backoff with errno 2
......
# All requested packages already installed.
# packages in environment at C:\Miniconda3\envs\35:
#
llvmlite                  0.13.0.dev              py35_61    numba

**Comment by pitrou** [comment]
```
(35) Z:\numba>conda info
Current conda install:

               platform : win-64
          conda version : 4.2.1
       conda is private : False
      conda-env version : 2.5.2
    conda-build version : 1.21.11+0.g5b44ab3.dirty
         python version : 3.5.1.final.0
       requests version : 2.10.0
       root environment : C:\Miniconda3  (writable)
    default environment : C:\Miniconda3\envs\35
       envs directories : C:\Miniconda3\envs
          package cache : C:\Miniconda3\pkgs
           channel URLs : defaults
            config file : C:\Users\Antoine\.condarc
           offline mode : False
```

**Comment by HugoTian** [comment]
First of all, there is a output bug, will be fix by PR #3304. When we do conda config --get channels, the top one has highest priority , and the bottom one has lowest priority. The output is reversed.

**Comment by HugoTian** [comment]
For your first ticket, default has high priority than numba, so the behavior of conda update --all is correct, and the conda update llvmlite has bug. The bug comes from [here](https://github.com/conda/conda/blob/4.2.1/conda/cli/install.py#L243), it just check version and build in this if case, which leads to "already satisfied" result. 

It will be fixed soon

**Comment by HugoTian** [comment]
For the second ticket, I have no idea. numba has higher priority, and it should not be replaced. May be it is related to the "Uncaught backoff with errno 2".

**Comment by HugoTian** [comment]
We will fix this issue and test it in PR #3306
